### ISSUE TITLE
Addon fails to initialize after pr #35

### ISSUE THREAD

**Issue by antonshusharin** [cartertemm/AI-content-describer#36]
Since commit a9ece1c, the addon fails to load with the following written to the log:
```
ERROR - globalPluginHandler.listPlugins (00:19:11.634) - MainThread (19128):
Error importing global plugin AIContentDescriber
Traceback (most recent call last):
  File "globalPluginHandler.pyc", line 23, in listPlugins
  File "importlib\__init__.pyc", line 126, in import_module
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Users\anton\AppData\Roaming\nvda\addons\aiContentDescriber\globalPlugins\AIContentDescriber\__init__.py", line 42, in <module>
    import model_configuration
  File "C:\Users\anton\AppData\Roaming\nvda\addons\aiContentDescriber\globalPlugins\AIContentDescriber\model_configuration.py", line 158, in <module>
    class Claude3_5SonnetConfigurationPanel(ClaudeConfigurationPanel):
  File "C:\Users\anton\AppData\Roaming\nvda\addons\aiContentDescriber\globalPlugins\AIContentDescriber\model_configuration.py", line 159, in Claude3_5SonnetConfigurationPanel
    model = description_service.Claude3_5Sonnet()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'description_service' has no attribute 'Claude3_5Sonnet'
```

Looks like this commit was based on an outdated version of the description_service.py file, which caused the class  for the Claude 3.5 Sonnet model to go missing. FYI @kefaslungu.

**Comment by kefaslungu** [comment]
Hello @antonshusharin,

My code didn't temper with that class, I only made edits to the Gemini class.

So I don't know what is rung with the Clawde class. Everything however is working well at my own end, waiting for an official release from addon author.

Or is there one I don't know of?

**Comment by antonshusharin** [comment]
@kefaslungu  I am running from source, from the main branch. Here's my guess as to what might have happened here:
- You were probably running the latest public release of the addon
- You took the description_service.py file from this distribution and made your Gemini-specific change. You then included this modified file in PR #35 
- What you likely weren't aware of was that since that release, other changes had been made in this file and others to add support for the new Claude 3.5 model. As a result, you were actually editing an outdated version of description_service.py, and when the pull request was merged the Claude changes in description_service.py were overwritten.
- As a result, there is now a mismatch in the repository where model_configuration.py references the Claude 3.5 Sonnet class from description_service.py, which is no longer present there.
- However, you are probably still running the latest release, which has no references to the 3.5 Sonnet model. This is why everything works fine on your end.

Please be more careful in the future and make sure your work is based on the latest version of the code.

**Comment by cartertemm** [comment]
Hey @antonshusharin,

Amazing catch, thank you!

I've restored support for 3.5 sonnet in 86d53. Feel free to give it a try and let me know whether that did the trick.

It's partially my bad for deciding it would be a mighty fine idea to review a PR on my phone on the train, saying LGTM, and hammering merge. Rookie mistake from someone who should definitely know better.

That being said, you are spot on to emphasize the importance in ensuring that all contributions start by checking out the main branch of the repository instead of the version from the releases tab or the NVDA add-ons site.

**Comment by antonshusharin** [comment]
Yes, the initialization issue has been fixed. Thank you. I noticed that you also refactored the caching decorator, which was the next thing I was going to report. However it looks like only the models from OpenAI have their process method decorated with it, so no other provider will cache atm.
Have you considered moving the caching logic out into processs_image? This should eliminate the possibility of such a mistake in the future.
